src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java

```diff
@@ -716,14 +716,15 @@ public void close() {
 		 */
 		private CompletableFuture<Void> receivePumpAsync() {
 			return receivePumpCoreAsync().handle((keepGoing, ex) -> {
-				if (keepGoing) {
+				if (keepGoing) {				
 					receivePumpAsync();
+				} else {
+					this.onOffline(ex);
+					if (ex != null) {
+						throw RelayLogger.throwingException(ex, this, TraceLevel.WARNING);
+					}
```

Comment:
Revert these order changes

Reply:
This much of the original is indeed correct: log the exception, then call onOffline.

---
In reply to: [257831721](https://github.com/Azure/azure-relay-java/pull/16#discussion_r257831721) [](ancestors = 257831721)